src/core/Akka.Persistence/Journal/AsyncWriteJournal.cs

```diff
@@ -315,9 +349,40 @@ protected Exception TryUnwrapException(Exception e)
                 if (aggregateException.InnerExceptions.Count == 1)
                     return aggregateException.InnerExceptions[0];
             }
+
             return e;
         }
 
+        // ValueDelegate implementation for Func<IPersistentRepresentation, Exception, object> mapper
+        private interface IMapperDelegate
+        {
+            object Invoke(WriteMessages message, IPersistentRepresentation rep, Exception ex);
+        }
+
+        private sealed class SuccessDelegate : IMapperDelegate
```

Comment:
To Fully ValueDelegate these, change to a struct and pass in `deault(SuccessDelegate)` instead of `SuccessDelegate.Instance`.

May want to test both ways, since that's where we'll be doing the Virt call cost vs code size tradeoff